### PR TITLE
149 implement document query service

### PR DESCRIPTION
**Pull request by fmatthies**

### CODE REVIEW

**Comment by fmatthies** [comment]
depending on top-api (I made a version increase there 0.9.1 -> 0.9.2) the version needs to be adjusted here as well; this is the conflict

File: src/main/java/care/smith/top/backend/service/nlp/DocumentService.java [hunk 1]
```diff
@@ -45,26 +45,68 @@ public DocumentService(
...
+      @NonNull String documentName, Integer page) {
+    return documentRepository
+        .findDocumentEntitiesByDocumentNameContains(documentName, pageRequestOf(page))
+        .map(DocumentEntity::toApiModel);
+  }
+
+  public Page<Document> getDocumentsByIds(
+      @NonNull Collection<String> ids, Integer page) {
+    //ToDo: something's not working with the repository constructed methods; so I needed to implement my own filtering and paging
+//    Page<DocumentEntity> documentPageEntity = documentRepository.findDocumentEntitiesByIdIn(ids, pageRequestOf(page));
```
Comment (ChristophB) [inline src/main/java/care/smith/top/backend/service/nlp/DocumentService.java:79]: In order for this to work correctly, I think the method name needs to be adjusted.

Something like: "findAllByIdIn"

The same applies to the other `DocumentRepository` methods.

**Review by ChristophB — CHANGES_REQUESTED** [review]

File: src/main/java/care/smith/top/backend/service/nlp/DocumentService.java (hunk 1 above)
Comment (fmatthies) [reply src/main/java/care/smith/top/backend/service/nlp/DocumentService.java:79]: Yeah, I thought so, but there is no "findAllByIdIn"; if I want something with "In" I only get this kind of method with "DocumentEntities" recommended

**Review by fmatthies — COMMENTED** [review]
all suggestions resolved as possible

**Review by ChristophB — APPROVED** [review]

**Pull request merged**